### DISCUSSION TITLE
Error in 'RenameBoundaryGenerator' when the 'new_boundary_id' is large

### DISCUSSION THREAD

**Discussion by echo1115** [idaholab/moose#22793] [Q&A Meshing] [answered]
Hi all,
When using `mesh-only ` to run the case of [moose/test/tests/meshgenerators/rename_boundary_generator/name_and_id.i ](https://github.com/idaholab/moose/blob/next/test/tests/meshgenerators/rename_boundary_generator/name_and_id.i) , I changed the `old_boundary` and `new_boundary` to `old_boundary_id` and `new_boundary_id` in the object [rename_both_id_and_name] :
```
  [rename_both_id_and_name]
    type = RenameBoundaryGenerator
    input = ed0
    old_boundary_id = '100' 
    new_boundary_id = '40000' #'20000' #'101'
  []
```
when the `new_boundary_id  `is 101 or 20000, the mesh was created successfully, 
but when setting it to 40000, it failed with the message:
```
Error writing sidesets
[0] ../src/mesh/exodusII_io_helper.C, line 2697, compiled Apr 13 2022 at 21:08:08

*** ERROR ***
Error writing sidesets

application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0

```

in the https://github.com/libMesh/libmesh/blob/devel/src/mesh/exodusII_io_helper.C, I find the error message but I don't understand：

```
      ex_err = exII::ex_put_sets(ex_id, side_boundary_ids.size(), sets.data());
      EX_CHECK_ERR(ex_err, "Error writing sidesets");
```
I hope your help, thanks!

**Comment by GiudGiud** [comment, marked as answer]
Hello

In libmesh you may find the integer type for boundary ids
```
50 #if LIBMESH_BOUNDARY_ID_BYTES == 1
  [ 51](https://libmesh.github.io/doxygen/namespacelibMesh.html#a3f4d8a52b5ef6ff6156ca209e12b5564) typedef int8_t [boundary_id_type](https://libmesh.github.io/doxygen/namespacelibMesh.html#a3f4d8a52b5ef6ff6156ca209e12b5564);
   52 #elif LIBMESH_BOUNDARY_ID_BYTES == 4
   53 typedef int32_t [boundary_id_type](https://libmesh.github.io/doxygen/namespacelibMesh.html#a3f4d8a52b5ef6ff6156ca209e12b5564);
   54 #elif LIBMESH_BOUNDARY_ID_BYTES == 8
   55 typedef int64_t [boundary_id_type](https://libmesh.github.io/doxygen/namespacelibMesh.html#a3f4d8a52b5ef6ff6156ca209e12b5564);
   56 #else // LIBMESH_BOUNDARY_ID_BYTES = 2 (default)
   57 typedef int16_t [boundary_id_type](https://libmesh.github.io/doxygen/namespacelibMesh.html#a3f4d8a52b5ef6ff6156ca209e12b5564);
   58 #endif
```

you can see that the defaut is an int16. The biggest int16 is 32767 so that's your problem.

We should probably catch this in MOOSE before trying to write a file though

Guillaume

**Reply by GiudGiud** [reply]
I ll raise an issue about this once I've narrowed it down to where it's failing. 

Thank you for reporting this

**Reply by GiudGiud** [reply]
btw new_boundary_id is deprecated. Please avoid using that. new_boundary will set the id and the name when using a number

**Reply by echo1115** [reply]
Thanks for your reply! @GiudGiud . If I want to use an int32 or bigger one, how should I set in MOOSE?

**Reply by GiudGiud** [reply]
You would need to build libmesh from source to do that, and modify it in libmesh.

**Reply by echo1115** [reply]
thanks for your suggestion. @GiudGiud
